### DISCUSSION TITLE
firewallcmd-ipset and logging

### DISCUSSION THREAD

**Discussion by ghost** [fail2ban/fail2ban#3815] [Q&A] [answered]
Hello!
I use firewallcmd-ipset action and I want to enable logging for that.
I tried to add firewallcmd-rich-logging. but the ports are not resolved here (I've discussed that on the mailing list, bot nobody responded).
https://sourceforge.net/p/fail2ban/mailman/message/58792169/

How can I configure logging for firewallcmd-ipset, so I can see the logs in the regular firewall logs?

Is that something that needs to be fixed by f2b or do I use it incorrectly?

**Comment by sebres** [comment, marked as answer]
> I've discussed that on the mailing list, bot nobody responded...

The development and issue-tracker moved to GH several years ago. Strange is that I didn't get any notification from this SF mailing-list. As often, something is wrong there, again.

> `$p` isn't resolved to the port definition.

Sure, since it is enclosed in single quotes...
It looks indeed like a bug, I guess here:
https://github.com/fail2ban/fail2ban/blob/216f0abb5eeee4581f1dd025b8cf909ab06ff559/config/action.d/firewallcmd-rich-rules.conf#L38

and must be changed like:
```diff
- fwcmd_rich_rule = rule family='<family>' source address='<ip>' port port='$p' protocol='<protocol>' %(rich-suffix)s
+ fwcmd_rich_rule = rule family='<family>' source address='<ip>' port port=\"$p\" protocol='<protocol>' %(rich-suffix)s 
```
However it looks like it was initially introduced incorrectly in 3d239215cd2681951d9bce20f2622fa0565136d6 (8 years ago!).
It seems that nobody really uses `firewallcmd-rich-*` actions... I guess even all the `firewall*` actions, due to native net-filter, what people use instead.

**Comment by sebres** [comment]
Fixed now in d4663e89414405752a682c676da1b48f8de85ab7.
